cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataAdmin.java

```diff
@@ -61,18 +66,43 @@ void addProperties(Id.NamespacedId entityId, Map<String, String> properties)
   Set<MetadataRecord> getMetadata(Id.NamespacedId entityId) throws NotFoundException;
 
   /**
+   * Returns a set of {@link MetadataRecord} representing all metadata (including properties and tags) for the specified
+   * {@link Id.NamespacedId}.
+   *
+   * @throws NotFoundException if the specified entity was not found
+   */
+  // TODO: Should this return a single metadata record instead or is a set of one record ok?
+  Set<MetadataRecord> getMetadata(MetadataScope scope, Id.NamespacedId entityId) throws NotFoundException;
+
+  /**
    * @return a {@link Map} representing the business metadata of the specified {@link Id.Application},
    * {@link Id.Program}, {@link Id.DatasetInstance} or {@link Id.Stream}
    * @throws NotFoundException if the specified entity was not found
    */
+  // TODO: This should perhaps return a Map<MetadataScope, Map<String, String>>
   Map<String, String> getProperties(Id.NamespacedId entityId) throws NotFoundException;
 
   /**
+   * @return a {@link Map} representing the business metadata of the specified {@link Id.Application},
+   * {@link Id.Program}, {@link Id.DatasetInstance} or {@link Id.Stream}
+   * @throws NotFoundException if the specified entity was not found
+   */
+  Map<String, String> getProperties(MetadataScope scope, Id.NamespacedId entityId) throws NotFoundException;
```

Comment:
Oh my comment was on the Map<> returned by get without scope. The case is similar to tags right ? We should return with scope because if the key in the user and system metadata is same it will lead to issue.